optuna/samplers/_tpe/multi_objective_sampler.py

```diff
@@ -247,8 +266,11 @@ def _split_mo_observation_pairs(
 
         # Solving HSSP for variables number of times is a waste of time.
         # We cache the result of splitting.
-        if _SPLITCACHE_KEY in trial.system_attrs:
-            split_cache = trial.system_attrs[_SPLITCACHE_KEY]
+        with self._split_cache_lock:
+            is_cached = trial._trial_id in self._split_cache
+        if is_cached:
+            with self._split_cache_lock:
+                split_cache = self._split_cache[trial._trial_id]
```

Comment:
How about adding test cases to validate the `split_cache` is appropriately set?

Reply:
I have added a simple test case for caches.